File: backend/models/wazuh.py
from pydantic import BaseModel, ConfigDict


class WazuhAlert(BaseModel):
    model_config = ConfigDict(extra='ignore')

    alert_id: str
    timestamp: str
    rule_id: str
    rule_description: str
    rule_level: int
    severity: str
    agent_id: str
    agent_name: str
    src_ip: str | None
    dst_ip: str | None
    mitre_technique: str | None
    mitre_tactic: str | None
    full_log: str
    groups: list[str]

File: packages/shared/types/wazuh.ts
export type WazuhSeverity = 'low' | 'medium' | 'high' | 'critical'

export interface WazuhAlert {
  alert_id: string
  timestamp: string
  rule_id: string
  rule_description: string
  rule_level: number        // 0-15 Wazuh scale
  severity: WazuhSeverity
  agent_id: string
  agent_name: string
  src_ip: string | null
  dst_ip: string | null
  mitre_technique: string | null
  mitre_tactic: string | null
  full_log: string
  groups: string[]
}
